internal/controllers/machine_controller.go

```diff
@@ -221,9 +235,20 @@ func (r *MachineReconciler) Start(ctx context.Context) error {
 }
 
 func (r *MachineReconciler) startCheckAndEnqueueVolumeResize(ctx context.Context, log logr.Logger) {
+	const name = "volume-size"
+	log = log.WithName(name)
+	opsDuration := metrics.OperationDuration.WithLabelValues(name)
+	opsErrors := metrics.OperationErrors.WithLabelValues(name)
```

Comment:
Can we keep passing the prepared logger and maybe define a global name like you did for `MachineReconcilerName`?

Reply:
about passing argument i am not completely sure.  I personally suggest remove passing over argument and directly initialize in function log from r.log. because passing of arguments doesn't add any benefits because start.. functions are methods of MachineReconciler and they have direct access to main logger. Now we just "spilitting" configuration of operation into 2 functions. I prefer keep it together.

My suggestion:

```
func (r *MachineReconciler) startCheckAndEnqueueVolumeResize(ctx context.Context) {
  log := r.log.WithName(MachineReconcilerVolumeSize)
  ...
```

Reply:
`WithName` adds context and appends the name to the existing name.

Reply:
I really don't understand why are you giving me here function description and you don't react usefully to my comment.
But okay it appends name separate with `.`, if you want to see whole description.
But it is useless, if we discuss how passing child logger into function.

Reply:
Dissecting the requests here.

1. Request to introduce global consts for the individual logger decorators: I agree with @lukasfrank here, being consistent here is introducing a const for each decorator name similar to https://github.com/ironcore-dev/libvirt-provider/pull/384/files#diff-a2d4663c7af1f98625f6b53f602e53ce910de934bd827f232e31db91bf9bb5acR49

2. Injecting vs. Self-decorating: This is a bit of a matter of taste in this particular spot. You can argue for both sides, keeping the injection by method argument enables the method to just use whatever logger is passed in and go with it. Also, configuration is kept outside of the method itself and is done by the calling instance. The self-decorating approach @lukas016 suggests is also valid as it encapsulates the responsibilities for behavior in the `start...` methods. Ultimately, the suggestion by @lukas016 is a good compromise:

```
func (r *MachineReconciler) startCheckAndEnqueueVolumeResize(ctx context.Context) {
  log := r.log.WithName(MachineReconcilerVolumeSize)
  ...
```

When testing, you pass in your mock logging provider to the containing reconciler struct and the `start...` methods consume this, with also self-contained decorators.